conceptnet5/db/query.py

```diff
@@ -1,156 +1,147 @@
 from conceptnet5.db.connection import get_db_connection
+from conceptnet5.db.config import DB_NAME
 from conceptnet5.edges import transform_for_linked_data
 import json
 import itertools
 from ftfy.fixes import remove_control_chars
 
-NODE_PREFIX_CRITERIA = {'node', 'other', 'start', 'end'}
 LIST_QUERIES = {}
 FEATURE_QUERIES = {}
 
-RANDOM_QUERY = "SELECT uri, data FROM edges TABLESAMPLE SYSTEM(0.01) ORDER BY random() LIMIT %(limit)s"
-RANDOM_NODES_QUERY = "SELECT * FROM nodes TABLESAMPLE SYSTEM(1) WHERE uri LIKE :prefix ORDER BY random() LIMIT %(limit)s"
-DATASET_QUERY = "SELECT uri, data FROM edges TABLESAMPLE SYSTEM(0.01) WHERE data->'dataset' = %(dataset)s ORDER BY weight DESC OFFSET %(offset)s LIMIT %(limit)s"
-
-
-TOO_BIG_PREFIXES = ['/c/en', '/c/fr', '/c/es', '/c/de', '/c/ja', '/c/zh',
-                    '/c/pt', '/c/la', '/c/it', '/c/ru' ,'/c/fi']
+if DB_NAME == 'conceptnet-test':
+    # Random queries sample 10% of edges; dataset queries sample 100% of edges.
+    # This makes sure that we get matches in the test database where there
+    # isn't much data.
+    #
+    # The TABLESAMPLE SYSTEM sampler gets 'chunks' of edges in a way that
+    # favors efficiency over independent sampling, so it's quite possible for
+    # it to skip every edge in a dataset in a correlated way unless we sample
+    # 100%.
+    RANDOM_QUERY = """
+        SELECT uri, data, weight FROM edges
+        TABLESAMPLE SYSTEM(10)
+        ORDER BY random() LIMIT %(limit)s
+    """
+else:
+    # In the real database, random queries sample 0.01% of edges, and dataset
+    # queries sample 1% of edges.
+    RANDOM_QUERY = """
+        SELECT uri, data, weight FROM edges
+        TABLESAMPLE SYSTEM(0.01)
+        ORDER BY random() LIMIT %(limit)s
+    """
 
+# A query that's optimized for producing the edges, grouped by feature, that
+# you get when you look up a concept in the Web interface.
 NODE_TO_FEATURE_QUERY = """
 WITH node_ids AS (
-    SELECT p.node_id FROM nodes n, node_prefixes p
-    WHERE p.prefix_id=n.id AND n.uri=%(node)s
-    LIMIT 10
```

Comment:
Why is there no limit here anymore?

Reply:
It was kind of a desperate optimization. If the 'node' matched too many things, we'd just return an arbitrary 10 of them.

Now that we're looking up the prefix and the features are stored separately for each prefix, we don't need to worry about this.